src/polly/src/event_manager.rs

```diff
@@ -284,21 +284,14 @@ impl EventManager {
         Ok(())
     }
 
-    /// Register a new event handler.
-    /// handler.init() will specify the pollable and event set.
-    ///
     /// Returns the handler wrapped in an Arc<Mutex>.
-    pub fn register<T: EventHandler + 'static>(&mut self, handler: T) -> Result<Arc<Mutex<T>>> {
-        let wrapped_handler = Arc::new(Mutex::new(handler));
-        self.register_protected(wrapped_handler.clone())?;
-        Ok(wrapped_handler)
+    pub fn wrap_handler<T: EventHandler + 'static>(handler: T) -> Arc<Mutex<T>> {
```

Comment:
Why does this function belong to the `EventManager`?

Reply:
Fixed in #1494